src/Adapter/MSTest.TestAdapter/Discovery/AssemblyEnumeratorWrapper.cs

```diff
@@ -114,6 +114,10 @@ private static ICollection<UnitTestElement> GetTestsInIsolation(string fullFileP
             PlatformServiceProvider.Instance.AdapterTraceLogger.LogWarning(Resource.OlderTFMVersionFound);
         }
 
-        return assemblyEnumerator.EnumerateAssembly(fullFilePath, warnings);
+        // This method runs inside of appdomain, when appdomains are available and enabled.
+        // Don't move the 'warning' parameter to be written into by reference, it will not return the warnings from appdomain.
+        AssemblyEnumerationResult result = assemblyEnumerator.EnumerateAssembly(fullFilePath);
+        warnings.AddRange(result.Warnings);
```

Comment:
nit: return AssemblyEnumerationResult here and get rid of the `warnings` parameter of `GetTestsInIsolation`. The caller is always passing a fresh list anyways, so I think it's easier to follow to just return the result here. The caller can add extra warnings to the returned list if needed

Reply:
done